src/Symfony/Component/Validator/Constraints/LengthValidator.php

```diff
@@ -38,9 +38,15 @@ public function validate($value, Constraint $constraint)
         }
 
         $stringValue = (string) $value;
+        $normalizedString = $stringValue;
+
+        $normalizedString = $constraint->ltrim ? preg_replace('/^[\pZ\pC]+/u', '', $normalizedString) : $normalizedString;
```

Comment:
This makes a strong supposition, namely that the input is in UTF-8. Is that guaranteed to be the case?

Reply:
Hi @nicolas-grekas you are right, thanks.
Do you believe that converting to ``UTF-8`` using the already existing ``charset`` option is a good approach (as the code below suggests)? Do you have any other/better idea?
```php
$stringValue = (string) $value;

if (!$invalidCharset = !@mb_check_encoding($stringValue, $constraint->charset)) {
    $normalizedString = mb_convert_encoding($stringValue, 'UTF-8', $constraint->charset);

    $normalizedString = $constraint->ltrim ? preg_replace('/^[\pZ\pC]+/u', '', $normalizedString) : $normalizedString;
    $normalizedString = $constraint->rtrim ? preg_replace('/[\pZ\pC]+$/u', '', $normalizedString) : $normalizedString;

    $normalizedString = $constraint->mergeConsecutiveWhitespaces ? preg_replace('/[\pZ\pC]+/u', ' ', $normalizedString) : $normalizedString;

    $length = mb_strlen($normalizedString, 'UTF-8');
}
```

Reply:
That's too heavy for a constraint IMHO.
I'd suggest sticking to regular spaces (ie use `ltrim` and a non unicode regex)
I'm also personally not convinced by the `mergeConsecutiveWhitespaces`, but maybe others see the use case

Reply:
Yeah, it also looks error prone somehow. Besides, PHP's ```trim``` functions behavior are well known and we should probably stick to it.
I will leave the ```mergeConsecutiveWhitespaces``` for another PR.

Reply:
@nicolas-grekas @renan-taranto I mentioned an use case in my issue - e.g. whitespace-styled forum posts that need to have enough content. 

As for "too heavy" - IMO validators are supposed to cover as many cases as possible. Unicode whitespaces are not uncommon these days. Not sure why you wouldn't cover them when it's possible to cover them.

Reply:
@Destroy666x I believe that someone using the ```mergeConsecutiveWhitespaces``` will have to implement a regex on the value (eg.: for persisting) after validating and it may yield different results than the validator, that would require knowledge about the constraint code and that's not good. PHP's ```trim``` functions are well known and wont cause that kind of confusion, for that same reason we should probably use them for the  ```ltrim``` and ```rtrim``` options.